pkg/k8s/provisioner/provisioner_controller.go

```diff
@@ -0,0 +1,128 @@
+		if string(b.Spec.Class) != pc.Name {
+			log.Info("provisioner", "found bundle name that does not reference the current provisioner class", b.Name)
+			continue
+		}
+		filtered = append(filtered, &b)
+	}
+	if len(filtered) == 0 {
+		log.Info("no bundles found specifying the current provisionerclass")
+		return ctrl.Result{}, nil
+	}
+
+	var errors []error
+	for _, bundle := range filtered {
+		log.Info("provisioner", "found bundle name that references the current provisionerclass", bundle.Name)
+		if err := r.ensureBundleVolume(ctx, bundle.GetName()); err != nil {
+			errors = append(errors, err)
+		}
+	}
+
+	return ctrl.Result{}, utilerrors.NewAggregate(errors)
+}
+
+// ensureBundleVolume is a helper method responsible for ensuring the bundle's status
+// references an existing volume that will the content stored in the bundle sources'
+// filesystem.
+func (r *Reconciler) ensureBundleVolume(ctx context.Context, bundleName string) error {
+	fresh := &v1alpha1.Bundle{}
+	if err := r.Client.Get(ctx, getNonNamespacedName(bundleName), fresh); err != nil {
+		return err
+	}
+
+	if fresh.Status.Volume != nil {
+		// TODO(tflannag): Check the status is still valid
+		// TODO(tlfannag): Handle case where the object was just updated and requeued
+		return nil
+	}
+
+	// TODO(tflannag): Avoid hardcoding -- likely can find the PVC by label selecting
+	pvc := &corev1.PersistentVolumeClaim{}
+	if err := r.Client.Get(ctx, types.NamespacedName{Name: "manifests", Namespace: r.globalNamespace}, pvc); err != nil {
+		return client.IgnoreNotFound(err)
+	}
+	fresh.Status.Volume = &corev1.LocalObjectReference{
+		Name: pvc.GetName(),
+	}
+
+	r.log.Info("volume", "attempting to update the bundle volume", fresh.GetName(), "with pvc name", pvc.GetName())
+	if err := r.Client.Status().Update(ctx, fresh); err != nil {
+		return err
+	}
+	r.log.Info("volume", "bundle status has been updated to point to a volume created", fresh.GetName())
+
+	return nil
+}
+
+// TODO(tflannag): This logic should be abstracted further away as a newVolume
+// type operation.
+// TODO(tflannag): Need a way to avoid run-once operations that cannot be reconciled
+// when state changes, e.g. avoid an InstallPlan-like scenario.
+// TODO(tflannag): Avoid hardcoding the storage requests.
+func (r *Reconciler) createPVC(bundleName string) (*corev1.PersistentVolumeClaim, error) {
+	pvcName := fmt.Sprintf("%s-pvc", bundleName)
+	pvc := &corev1.PersistentVolumeClaim{}
+	ctx := context.Background()
+
+	nn := types.NamespacedName{Name: pvcName, Namespace: r.globalNamespace}
+	if err := r.Client.Get(ctx, nn, pvc); err != nil {
+		if !apierrors.IsNotFound(err) {
+			return nil, err
+		}
+
+		pvc.SetName(pvcName)
+		pvc.SetNamespace(r.globalNamespace)
+		volumeMode := corev1.PersistentVolumeFilesystem
+
+		pvc.Spec = corev1.PersistentVolumeClaimSpec{
+			VolumeMode:  &volumeMode,
+			AccessModes: []corev1.PersistentVolumeAccessMode{corev1.ReadWriteOnce},
```

Comment:
I know we are still thinking about this, but is there a reason we don't use `ReadOnlyMany` AccessMode [1] for the PVC? Otherwise we can't share the content among multiple nodes. 

[1] https://kubernetes.io/docs/concepts/storage/persistent-volumes/#access-modes

Reply:
Short answer: I'm developing locally on a single node cluster (e.g. kind) where it doesn't really matter too much what type of access mode(s) I'm using when creating a PVC, and I'm mainly interested to prototyping the ideas outlined in the e2e strawman.

I think looking forward, we'd like to abstract away the storage backend as much as possible. This could mean unpacked bundle contents are housed in-or-off cluster, as a PVC or shared across ConfigMaps (the latter being a little tricker for introspection of resources prior to approval), a database, or alternatives I haven't yet wrapped my head around. You might be able to make the argument that specification of a storage backend should be decided by the provisioner, maybe even defined as a parameter in a ProvisionerClass resource.

One other quick note, and this was especially true in the past and may not hold as much weight now, is that ReadWriteMany PVCs are typically harder to create as there are fewer storage provisioners that support that type of access mode.